### ISSUE TITLE
slackroll upgrade does not catch new patched kernel version on stable

### ISSUE THREAD

**Issue by Gunbird** [rg3/slackroll#53]
After upgrading all packages, I noticed that a more recent kernel existed in the remote tree but slackroll kernel-upgrade did not catch it.

```
slackroll list-versions kernel-generic
Available versions:
    kernel-generic:
        Local:  kernel-generic-5.15.117-x86_64-1
        Remote: ./patches/packages/linux-5.15.139/kernel-generic-5.15.139-x86_64-1.txz
        Remote: ./patches/packages/old-linux-5.15.117/kernel-generic-5.15.117-x86_64-1.txz

End of list
```

But when I try to upgrade : 

```
slackroll kernel-upgrade 
No outdated kernel packages
Total size: 0b
Choose option:
    (S) Shell
    (X) Done
```

**Comment by aclemons** [comment]
Thanks for opening this. I also noticed this not flagged for upgrade when it was added to the changelog.

The reason for this is because the directory `slackware64-15.0/patches/packages` on the mirrors has both `old-linux-5.15.117` and `linux-5.15.139`.

The remote list logic is here:

https://github.com/slackroll/slackroll/blob/v51/slackroll#L594-L616

It finds two packages in `patches` and removes the one in main as expected.

The outdated check then does this:

https://github.com/slackroll/slackroll/blob/v51/slackroll#L1625-L1629

Since the currently installed kernel version is 5.15.117 and that is in the list of versions returned by `extend_remote_list`, the package is marked as "up to date" (len > 0).

I added `old-linux` to my blacklist to ignore the old patch, but maybe slackroll should handle this natively. I can't recall Pat ever keeping the old patch like this before and he did not explicitly mention it in the changelog entry. I don't think this was a mistake though.

**Comment by rg3** [comment]
Interesting. I don't follow Slackware closely anymore, but maybe you can email Pat about the intention of old-linux, and if it's safe for users to run that. Normally, stable kernel updates include security fixes, so I'm not sure what the intention of having two 5.15.x kernel versions is here.

**Comment by aclemons** [comment]
Ok, this seems to be a new way Pat is handling kernel patches by keeping the old versions around, I guess in case there are regression for anyone.

https://slackware.osuosl.org/slackware64-15.0/patches/packages/

Now has both the previous patches 5.15.117 and 5.15.139.

I'll look at adding a filter for these by default.

**Comment by rg3** [comment]
> Ok, this seems to be a new way Pat is handling kernel patches by keeping the old versions around, I guess in case there are regression for anyone.

Alright, after seeing the directory contents I agree the sensible option is, as you suggested, ignoring /patches/packages/old-linux-* by default. IIRC, slackroll's kernel commands allow you to keep the old kernel around in case you find regressions, so that functionality is already covered and there's not much to gain by writing more complex logic around this. Just pretend those directories do not exist. :shrug:

**Comment by aclemons** [comment]
Sorry, I had this patch lying around but had not pushed it up and made a release..

I'll cut a new release with this fix and the one for #55 soon.
